pallets/parachain-staking/src/benchmarks.rs

```diff
@@ -2484,6 +2673,79 @@ mod benchmarks {
 		Ok(())
 	}
 
+	#[benchmark]
+	fn migrate_locks_to_freezes_batch(x: Linear<1, 100>) -> Result<(), BenchmarkError> {
+		// x is the number of accounts to migrate in the batch
+		use crate::MigratedDelegators;
+		use frame_benchmarking::whitelisted_caller;
+
+		let mut seed = Seed::new();
+		let mut delegator_accounts = Vec::new();
+
+		// Create a collator that delegators will delegate to
+		let collator = create_funded_collator::<T>(
+			"collator",
+			seed.take(),
+			min_candidate_stk::<T>(),
+			true,
+			1,
+			true,
+		)?;
+
+		// Create x delegator accounts with existing locks to migrate
+		for i in 0..x {
+			// Add extra amount to ensure each delegation is unique and larger than previous ones
+			// This prevents hitting the CannotDelegateLessThanOrEqualToLowestBottomWhenFull error
+			let extra_amount = BalanceOf::<T>::from(i.saturating_mul(100u32));
+			let delegator = create_funded_delegator::<T>(
+				"delegator",
+				seed.take(),
+				extra_amount,
+				collator.clone(),
+				false,
+				i + 1, // delegation count
+				true,
+			)?;
+
+			// Remove from MigratedDelegators to ensure it's not already marked as migrated
+			<MigratedDelegators<T>>::remove(&delegator);
+
+			// Calculate the actual delegation amount (min_delegator_stk + extra_amount)
+			let delegation_amount = min_delegator_stk::<T>() + extra_amount;
+
+			// Set an old-style lock on the account to simulate pre-migration state
+			T::Currency::set_lock(
+				DELEGATOR_LOCK_ID,
+				&delegator,
+				delegation_amount,
+				frame_support::traits::WithdrawReasons::all(),
+			);
+
+			delegator_accounts.push(delegator);
+		}
+
+		let caller: T::AccountId = whitelisted_caller();
+		let is_collator = false; // We're migrating delegators
```

Comment:
Are we sure that migrating a delegator is always worse than migrating a collator? Especially about proof size?